lib/logstash/inputs/stomp.rb

```diff
@@ -103,7 +95,18 @@ def subscription_handler
 
   public
   def run(output_queue)
+    # Handle disconnects
+    @client.on_connection_closed {
+      self.connect
+      subscription_handler # is required for re-subscribing to the destination
+    }
```

Comment:
actually this could stay in the register?

Reply:
unless we connected at least once we should not hit the callback?

Reply:
You're right that this could stay in the register. I moved this here only because of the `self.connect` line. I didn't even want the appearance of a `connect` in the `register` method.

If you feel strongly about this moving back I'm fine to do this, it was kind of a style thing for me.

Reply:
I don't feel strongly about it, I don't know the library, if this could be triggered before a connect :)